common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -109,6 +129,40 @@ public static void deleteRecursively(File file) throws IOException {
     }
   }
 
+  private static void deleteRecursivelyUsingUnixNative(File file) throws IOException {
+    final ProcessBuilder builder = new ProcessBuilder("rm", "-rf", file.getAbsolutePath());
+    Process process = null;
+    int exitCode = -1;
+
+    try {
+      // Merge error stream of the process with its stdout
+      // Later stdout to be consumed using {@link Process#getInputStream()}
+      builder.redirectErrorStream(true);
+
+      process = builder.start();
+
+      // In order to avoid deadlocks, consume the stdout (and stderr) of the process
+      final BufferedInputStream in = new BufferedInputStream(process.getInputStream());
```

Comment:
This works, but in Java 7 can we just redirect to `new File("/dev/null")` because we know this is a Unix OS? Might be simpler. PS I don't think we generally put `final` on vars unless it's necessary.